src/main/java/org/takes/facets/auth/social/PsGoogle.java

```diff
@@ -195,7 +195,12 @@ private static Identity parse(final JsonObject json) {
         } else {
             props.put("picture", image.getString("url", "#"));
         }
-        props.put("name", json.getString("displayName", "unknown"));
+        if (json.containsKey("displayName")
```

Comment:
@raphaelln try to reproduce the bug in a unit test first. you will see that your solution actually is not a solution